src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,170 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.*;
+import io.opentracing.propagation.Format;
+
+import javax.imageio.spi.ServiceRegistry;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The <code>GlobalTracer</code> class is not a {@link Tracer} implementation of itself, but instead a utility-class
+ * to centrally reference the {@link #tracer() singleton instance} of the configured {@link Tracer} implementation.
+ * <p>
+ * The singleton instance can be instantiated in one of two ways:
+ * <ol>
+ * <li>Explicitly, by calling {@link #register(Tracer)} with a configured tracer implementation, or:</li>
+ * <li>Automatically by using the Java <code>ServiceLoader</code> SPI mechanism to load an implementation
+ * from the classpath.</li>
+ * </ol>
+ *
+ * @author Sjoerd Talsma
+ * @see Tracer
+ */
+public final class GlobalTracer {
```

Comment:
note that this type in concept is similar to finagle's `Trace` which allows static trace functions which internally delegate to one or more tracers. probably sensible to review that if not already
https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

Finagle's propagation approach is different because their tracing approach is different. Users don't interact with spans, rather they give commands to affect them. So, Finagle propagates a trace context (with their normal system), and that context is used as a lookup key when sending a mutation event. This means tracers have a mapping between contexts and the spans associated with them. In short, users don't interact directly with a span object, rather when they "add a tag" they are really saying `inFlight.get(context).addTag(tag)`